controllers/ArticleController.js

```diff
@@ -0,0 +1,174 @@
+
+/** Class representing controller for articles. */
+class ArticleController {
+  /**
+     * Create an article for a user
+     * @param {object} req - The request object
+     * @param {object} res - The response object sent to user
+     * @return {object} A object containing created articles.
+     */
+  static createArticle(req, res) {
+    const {
+      title, description, body, tagList, imageUrl,
+    } = req.body.article;
+
+    const { userId } = req;
+
+    const articleObject = {
+      title, description, body, tagList, imageUrl, userId
+    };
+    /** check if image was provided in the request
+     * upload the image to cloudinary, save the article
+     * with the cloudinary URL in database but if an error
+     * was encountered from cloudinary go ahead and create the article
+     */
+    if (imageUrl) {
+      return cloudinary.v2.uploader.upload(imageUrl, { tags: 'basic_sample' })
+        .then(image => createArticleHelper(res, articleObject, image.url))
+        .catch(() => createArticleHelper(res, articleObject));
+    }
+
+    /** if there no image was provided go ahead to create the article */
+    return createArticleHelper(res, articleObject);
+  }
+
+  /**
+   * get an article using slug as query parameter
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} - the found article from database or error if not found
+   */
+  static getArticle(req, res) {
+    const { slug } = req.params;
+
+    return Article
+      .findOne({
+        where: { slug, },
+        include: [{
+          model: User,
+          attributes: { exclude: ['id', 'email', 'hashedPassword', 'createdAt', 'updatedAt'] }
+        }],
+        attributes: { exclude: ['id', 'userId'] }
+      })
+      .then((article) => {
+        /** if the article does not exist */
+        if (!article) {
+          return res.status(404).json({
+            errors: {
+              body: [
+                'Ooops! the article cannot be found.'
+              ]
+            }
+          });
+        }
+
+        return res.status(200).json({ article });
+      })
+      .catch(() => res.status(501).send('oops  seems there is an error find the article'));
+  }
+
+  /**
+   * get all articles created
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} - the found article from database or empty if not found
+   */
+  static listAllArticles(req, res) {
+    return Article
+      .findAll({
+        include: [{
+          model: User,
+          attributes: { exclude: ['id', 'email', 'hashedPassword', 'createdAt', 'updatedAt'] }
+        }],
+        attributes: { exclude: ['id', 'userId'] }
+      })
+      .then((articles) => {
+        /** check if there was no article created */
+        if (articles.length === 0) {
+          return res.status(200).json({
+            message: 'Your request was successful but no articles created',
```

Comment:
I suggest you just say "No articles found", as this is just a direct information to the user trying to retrieve articles

Reply:
Okay, yes but aren't we saying the same thing? To colour it, a user wants to get all the articles, the request was successful but there are no articles in the database yet.